### ISSUE TITLE
Adding some buttons

### ISSUE THREAD

**Issue by Atlas1394** [processhacker/processhacker#886]
I use PH for different purposes, mainly for games and monitoring, it would be nice for your application to add some "buttons" for activating/deactivating different features. For example when i play games, 90% of the cases I'm not interested in PH tracking performance statistics in background (consuming CPU cycles), but only in saved processes priority that PH applies constantly. Another example is the opposite, when i open PH just to see statistics and open processes, and I'm not interested on applying saved priorities (this is done automatically by PH at startup). I think that adding some global buttons for activating/deactivating monitoring, or activating/deactivating saved priority, would be nice, because it gives to the user the freedom of choice, depending on what he need. Sorry for my bad english, I hope it helps you with your great job in PH

**Comment by dmex** [comment]
> when i play games, 90% of the cases I'm not interested in PH tracking performance statistics in background (consuming CPU cycles)

I've been play games for years and never had any issues with PH running. We're talking about 0.10% (under 1 percent) to 0.01% when PH is minimized and only 2-3 function calls since all the other calls are from UI updates.

You can always pause the updating using F6 or via the View > Refresh Automatically menu which will pause everything.... I could insteam add a new option that automatically pauses all functionality when you're running a full screen application?

> not interested on applying saved priorities (this is done automatically by PH at startup)

I could add a button but you would have to start PH first to activate the button and those priorities would already be applied by the time you clicked the button so it wouldn't be very useful.

You should instead use the newer IFEO menus added to Process Hacker recently that use built-in Windows functionality for applying saved priorities since they're only applied when that program starts and only done once:

![image](https://user-images.githubusercontent.com/1306177/117539890-e96be780-b04f-11eb-95c5-9487bc5e211f.png)

**Comment by Atlas1394** [comment]
> when PH is minimized and only 2-3 function calls since all the other calls are from UI updates

yes, the good thing of PH is that it consumes very few cycles, simply my suggestion was just to slightly improve this a little more for games case, because when you play, you are only interested on PH detection of new processes in background ad apply the saved priority, not in GUI update or calculating statistics (RAM CPU DISK NETWORK etc). It's just a suggestion that I found nice, it's your choice to add it or not, of course.

> You can always pause the updating using F6 or via the View > Refresh Automatically menu which will pause everything

That's the point, I do not want to pause everything, but just the statistics. If a new process in background starts while I play, I want that PH detects it and set the specified priority, if I press F6 I'll stop everything, even the new process detection, saved priorities will not be applied.

> I could add a button but you would have to start PH first to activate the button and those priorities would already be applied by the time you clicked the button so it wouldn't be very useful

That's what I want, the priorities will be set when I click the button. It's just an additional feature to prevent that PH set automatically the priorities when it starts, if I do not want it (of course for example you can add a checkbox in which user specify to automatically set priorities on PH startup, as it does now)

I hope it has been explained well and that it helps the growth of this program!
